### PR TITLE
Review of 30/360 day count conventions

### PR DESCRIPTION
**Pull request by davidsteiner**
- Added test cases from QuantLib.
- Removed the `Option` wrapper around `market`.

### CODE REVIEW

File: src/time/daycounters/thirty360.rs [hunk 1]
```diff
@@ -146,19 +151,27 @@ impl Thirty360 {
...
+    fn default() -> Self {
+        Self {
+            market: Thirty360Market::USA,
+        }
+    }
+}
+
 #[typetag::serialize]
 impl DayCounters for Thirty360 {
     fn day_count(&self, d1: NaiveDate, d2: NaiveDate) -> i64 {
```
Comment (davidsteiner) [inline src/time/daycounters/thirty360.rs:164]: @thejerwang I'm of two minds whether this should return a signed integer? I feel like `Result<u64>` may be more appropriate (with `end_date < start_date` returning `Err`), although I see that QuantLib also returns signed integers. Thoughts on this?

File: Cargo.toml [hunk 2]
```diff
@@ -20,3 +20,6 @@ serde_json = "^1.0.107"
 strum = "0.25.0"
 strum_macros = "^0.25.3"
 typetag = "0.2.13"
+
+[dev-dependencies]
+rstest = "0.18"
```
Comment (ste645) [inline Cargo.toml:25]: Fixtures, learnt something new

Comment (davidsteiner) [reply Cargo.toml:25]: Yeah, although I feel like all rust test libs leave much to be desired..

`rstest` works well for parametrisation, though. After I gave Claude a crash course on converting things to `rstest` macros, it converted all this seamlessly 😄 
https://github.com/lballabio/QuantLib/blob/master/test-suite/daycounters.cpp#L849

Comment (ste645) [reply Cargo.toml:25]: yh Claude is a stand up guy

**Pull request merged**